sqlmesh/core/context.py

```diff
@@ -1867,10 +1883,13 @@ def print_info(self, skip_connection: bool = False, verbose: bool = False) -> No
                 self.config.get_state_connection(self.gateway), self.console, "State Connection"
             )
 
-        self._try_connection("data warehouse", self._engine_adapter.ping)
+        self._create_engine_adapters()
```

Comment:
Why do we call this here? Didn't we already call this in constructor?

Reply:
this is to test all engine adapter connections in the `config.yaml` when calling `sqlmesh info`. so the engine adapters wouldn't be in the context if not calling the create_engine_adapters first

Reply:
So the `create_engine_adapters `is called here to create and test **all** engine adapters, while before it is called to create only the ones that are needed for the project (ie the default adapter and the ones that are explicitly set in a model)

Reply:
Can we just load all available adapters once on load? I'm a little concerned with the control flow here. If I'm using Context in the notebook or a script, the `self. _engine_adapters` changes underneath me depending on which method I call.